src/main/groovy/org/transmartproject/batch/startup/JobDetails.groovy

```diff
@@ -0,0 +1,284 @@
+package org.transmartproject.batch.startup
+
+import com.google.common.base.Function
+import com.google.common.collect.Maps
+import groovy.transform.TypeChecked
+import groovy.transform.TypeCheckingMode
+import groovy.util.logging.Slf4j
+import org.springframework.batch.core.JobParameter
+import org.springframework.batch.core.JobParameters
+import org.transmartproject.batch.backout.BackoutJobSpecification
+import org.transmartproject.batch.clinical.ClinicalJobSpecification
+import org.transmartproject.batch.gwas.GwasJobSpecification
+import org.transmartproject.batch.highdim.acgh.data.AcghDataJobSpecification
+import org.transmartproject.batch.highdim.acgh.platform.AcghAnnotationJobSpecification
+import org.transmartproject.batch.highdim.metabolomics.data.MetabolomicsDataJobSpecification
+import org.transmartproject.batch.highdim.metabolomics.platform.MetabolomicsAnnotationJobSpecification
+import org.transmartproject.batch.highdim.mrna.data.MrnaDataJobSpecification
+import org.transmartproject.batch.highdim.mrna.platform.MrnaAnnotationJobSpecification
+import org.transmartproject.batch.highdim.proteomics.data.ProteomicsDataJobSpecification
+import org.transmartproject.batch.highdim.proteomics.platform.ProteomicsAnnotationJobSpecification
+import org.transmartproject.batch.highdim.rnaseq.data.RnaSeqDataJobSpecification
+import org.transmartproject.batch.highdim.rnaseq.platform.RnaSeqAnnotationJobSpecification
+import org.transmartproject.batch.i2b2.I2b2JobSpecification
+import org.transmartproject.batch.support.StringUtils
+import org.transmartproject.batch.tag.TagsLoadJobSpecification
+
+import java.nio.file.Files
+import java.nio.file.Path
+
+/**
+ * Gets all details needed for running a job.
+ */
+@TypeChecked
+@Slf4j
+final class JobDetails {
```

Comment:
I don't like the name change. `JobDetails` is very vague. If this is indeed intended to represent the "details needed for running a job", then a more specific name like `JobStartupDetails` or `JobInitializationData` would be better.